core/src/services/moka/backend.rs

```diff
@@ -31,29 +39,61 @@ use crate::*;
 impl Configurator for MokaConfig {
     type Builder = MokaBuilder;
     fn into_builder(self) -> Self::Builder {
-        MokaBuilder { config: self }
+        MokaBuilder {
+            config: self,
+            ..Default::default()
+        }
     }
 }
 
+type Weigher<K, V> = Box<dyn Fn(&K, &V) -> u32 + Send + Sync + 'static>;
+
+type AsyncEvictionListener<K, V> =
+    Box<dyn Fn(Arc<K>, V, RemovalCause) -> ListenerFuture + Send + Sync + 'static>;
+
 /// [moka](https://github.com/moka-rs/moka) backend support.
 #[doc = include_str!("docs.md")]
-#[derive(Default, Debug)]
+#[derive(Default)]
 pub struct MokaBuilder {
     config: MokaConfig,
+    expiry: Option<ExpiryWrapper<String, typed_kv::Value>>,
```

Comment:
Instead of providing all those options, how about accepting a `CacheBuilder` instead? This way, opendal can offer some basic options, but if users want more control, they can use the builder directly.

Reply:
If you think it's OK, would it be better for us to delete the other configuration methods? or just keep them to avoid breaking changes?

Reply:
I'm a bit hesitant to expose `typed_kv::Value` as it is. We currently have plans to remove both `typed_kv` and `kv` entirely.

If you're not in a hurry to use this, perhaps we can wait a bit until we're sure which direction to take.

Reply:
I can copy this code into my project now, so I'm not in a hurry.

Reply:
I did a refactor in https://github.com/apache/opendal/pull/6300

I think we can expose a `MokaCacheBuilder<String, MokaValue>` to users.

Reply:
Do you think we should expose other methods (like `name`, `time_to_live`, etc) to modify the moka cache builder besides the `configure` method?

Reply:
> Do you think we should expose other methods (like `name`, `time_to_live`, etc) to modify the moka cache builder besides the `configure` method?

Yes, I think we can keep the existing settings since they are much more convenient for our bindings users.